R/parse.R

```diff
@@ -25,7 +25,12 @@ parse_text <- function(text, registry = default_tags(), global_options = list())
 }
 
 parse_blocks <- function(file, env, registry, global_options = list()) {
-  parsed <- parse(file = file, keep.source = TRUE)
+
+  lines <- readLines(file, warn = FALSE, encoding = "UTF-8")
```

Comment:
Thanks. Find it needs to explicitly set the encoding of `srcfile` to preserve the `srcref` attribution:

```
  con <- file(file, encoding = "UTF-8")
  on.exit(close(con), add = TRUE)
  parsed <- parse(con, keep.source = TRUE, srcfile = srcfile(file, "UTF-8"))
```